src/test/java/com/commercetools/sync/commons/utils/RetryableSphereClientWithExponentialBackoffTest.java

```diff
@@ -0,0 +1,256 @@
+import static com.commercetools.sync.commons.utils.SphereClientConfigOptions.INITIAL_RETRY_DELAY;
+import static com.commercetools.sync.commons.utils.SphereClientConfigOptions.MAX_PARALLEL_REQUESTS;
+import static com.commercetools.sync.commons.utils.SphereClientConfigOptions.MAX_RETRIES;
+import static io.sphere.sdk.client.TestDoubleSphereClientFactory.createHttpTestDouble;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+class RetryableSphereClientWithExponentialBackoffTest {
+    private static final long TIMEOUT = 20000;
+    private static final TimeUnit TIMEOUT_TIME_UNIT = TimeUnit.MILLISECONDS;
+
+    @Test
+    void createClient_WithConfig_ReturnsSphereClient() {
+        final SphereClientConfig clientConfig =
+                SphereClientConfig.of("project-key", "client-id", "client-secret");
+        final SphereClient sphereClient = RetryableSphereClientWithExponentialBackoff.of(clientConfig);
+
+        assertThat(sphereClient instanceof SphereClient).isTrue();
+
+        assertThat(sphereClient.getConfig().getProjectKey()).isEqualTo("project-key");
+    }
+
+    @Test
+    void createClient_WithConfig_ReturnsBlockingSphereClient() {
+        final SphereClientConfig clientConfig =
+            SphereClientConfig.of("project-key", "client-id", "client-secret");
+        final SphereClient sphereClient = RetryableSphereClientWithExponentialBackoff.of(clientConfig, TIMEOUT, TIMEOUT_TIME_UNIT);
+
+        assertThat(sphereClient instanceof BlockingSphereClient).isTrue();
+
+        assertThat(sphereClient.getConfig().getProjectKey()).isEqualTo("project-key");
+    }
+
+    @Test
+    void createClient_withRetryDecorator_ShouldRetryWhen502HttpResponse() {
+        final SphereClient mockSphereUnderlyingClient =
+            spy(createHttpTestDouble(intent -> HttpResponse.of(HttpStatusCode.BAD_GATEWAY_502)));
+
+        final long maxRetryAttempt = 1L;
+        final Function<RetryContext, Duration> durationFunction = retryContext -> Duration.ofSeconds(1);
+
+        final SphereClient decoratedSphereClient =
+            decorateSphereClient(mockSphereUnderlyingClient, maxRetryAttempt, durationFunction, MAX_PARALLEL_REQUESTS);
+
+        final CustomerUpdateCommand customerUpdateCommand = getCustomerUpdateCommand();
+
+        assertThat(decoratedSphereClient.execute(customerUpdateCommand))
+            .failsWithin(2, TimeUnit.SECONDS) // first retry will be in 1 second, see: durationFunction.
+            .withThrowableOfType(ExecutionException.class)
+            .withCauseExactlyInstanceOf(BadGatewayException.class)
+            .withMessageContaining("502");
+
+        // first request + retry.
+        verify(mockSphereUnderlyingClient, times(2)).execute(customerUpdateCommand);
+    }
+
+    @Test
+    void createClient_Simulate_Stuck_SDK() {
```

Comment:
Doesn't look like a test to me. There's no assertion here.

Reply:
Removed.